### DISCUSSION TITLE
[Ark] [React] Potential Support React 17 (or smaller)

### DISCUSSION THREAD

**Discussion by t-keshi** [chakra-ui/ark#1329] [General]
`@ark-ui/react` currently support only React 18.
It seems to me that the main reason why it is not available on React 17 is we need `useId` of React18 for  implementation of a11y.
I think `@radix-ui/react-id` is  elegant approach to solve this problem.
How do you think?

https://github.com/radix-ui/primitives/tree/main/packages/react/id

I'm a big fun of ChakraUI and Ark, Thank you for great repository!

**Comment by gperdomor** [comment]
It would be nice to have React 16/17 support 🙏🏻

**Comment by cschroeter** [comment]
@t-keshi 

Thanks for bringing this up. Unfortunately it is not just `useId`. There a couple of other hooks used in Zag.js like `use` or `useSyncExternalStore` that does make this change not as easy.
